File: LineAutobotEmbeddings/gcp-cloud-function/js/storageService.js
const { Storage } = require('@google-cloud/storage');

class StorageService {
  constructor() {
    this.storage = new Storage();
  }

  /**
   * Writes a JSON object to a file in Google Cloud Storage
   * @param {string} bucketName - The name of the GCS bucket.
   * @param {string} path - The path to the directory where the file will be stored.
   * @param {string} fileName - The name of the file to be written.
   * @param {Object} data - The JSON object to write.
   * @returns {Promise<void>}
   */
  async writeJsonToFile(bucketName, path, fileName, data) {
    try {
      const fullPath = `${path}/${fileName}`;
      const file = this.storage.bucket(bucketName).file(fullPath);
      const jsonData = JSON.stringify(data, null, 2);
      await file.save(jsonData);
      console.log(`File ${fullPath} written to bucket ${bucketName}.`);
    } catch (error) {
      console.error('Error writing file:', error);
      throw new Error('Error writing JSON file');
    }
  }
}

module.exports = StorageService;

File: line_autobot_embeddings/services/cloud/gcp/storage_service.py
from google.cloud import storage
import json

class StorageService:
    def __init__(self):
        # Initialize Google Cloud Storage client
        self.storage_client = storage.Client()

    """
    Write a JSON object to a file in Google Cloud Storage
    :param bucket_name: GCS bucket name
    :param path: Directory path where the file will be stored
    :param file_name: Name of the file to write
    :param data: JSON data to write to the file
    """
    def write_json_to_file(self, bucket_name, path, file_name, data):
        try:
            # Generate full path
            full_path = f"{path}/{file_name}"
            
            # Get references to the bucket and blob
            bucket = self.storage_client.bucket(bucket_name)
            blob = bucket.blob(full_path)
            
            # Convert data to JSON string and upload it
            blob.upload_from_string(
                data=json.dumps(data),
                content_type='application/json'
            )
            print(f"File {file_name} written to {bucket_name}/{full_path}.")
        except Exception as e:
            print(f"Error writing file: {e}")
            raise RuntimeError('Error writing JSON file')